regenerate_sqlx_cache.sh

```diff
@@ -25,31 +25,29 @@ set -e  # Exit on any error
 
 echo "🔄 Regenerating SQLx query cache..."
 
-# Create a temporary directory for the database
-TEMP_DIR=$(mktemp -d)
-TEMP_DB="$TEMP_DIR/temp_sqlx_cache.sqlite"
+WORKSPACE_ROOT="$(pwd)"
+
+# Use shared temporary database in workspace root
+TEMP_DB="$WORKSPACE_ROOT/tempdb.sqlite"
 DATABASE_URL="sqlite:$TEMP_DB"
 
-echo "📁 Using temporary database: $TEMP_DB"
+rm -f "$TEMP_DB"
 
-# Function to cleanup on exit
-cleanup() {
-    echo "🧹 Cleaning up temporary files..."
-    rm -rf "$TEMP_DIR"
-}
-trap cleanup EXIT
+echo "📁 Using temporary database: $TEMP_DB"
 
 # Export DATABASE_URL for sqlx commands
 export DATABASE_URL
 
 echo "🗄️  Creating database..."
 cargo sqlx database create
 
-echo "🔄 Running migrations..."
-cargo sqlx migrate run
+for dir in swap monero-sys monero-rpc-pool; do
+    echo "🔄 Running migrations in $dir..."
```

Comment:
since we didn't include the deletion of the .sqlx directories in the crates (swap, monero-sys, ...) we should add `rm -rf .sqlx` here, too to make sure we get rid of it anyway